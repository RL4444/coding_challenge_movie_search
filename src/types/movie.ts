export interface IMovieApi {
    Poster: string;
    Title: string;
    Type: string;
    Year: string;
    imdbID: string;
}

interface IApiSuccessResponse {
    Error?: string;
    Response: string;
    Search?: Array <IMovieApi>;
    totalResults?: string;
}

export interface IApiResponse {
    error?: string;
    loading: boolean;
    response: IApiSuccessResponse;
}
