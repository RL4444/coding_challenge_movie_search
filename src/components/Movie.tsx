import React, { FC } from 'react';
import styled from 'styled-components';

import { IMovieApi } from '../types/movie';

const Container = styled.a`
    width: 100%;
    margin-top: 1em;
    display: flex;
    background: #262626;
    -webkit-box-shadow: 5px 5px 15px -6px rgba(0, 0, 0, 0.41);
    box-shadow: 5px 5px 15px -6px rgba(0, 0, 0, 0.41);
    padding: 10px 20px;
    @media (min-width: 800px) {
        border-radius: 6px;
        border-top-left-radius: 0px;
        border-bottom-left-radius: 0px;
        border-left: 3px solid ${(p) => p.theme.colors.white};
        transition-duration: 0.2;
        &:hover {
            border-left: 3px solid ${(p) => p.theme.colors.secondary};
            transition-duration: 0.3;
            -webkit-box-shadow: 5px 5px 15px -6px rgba(0, 0, 0, 0.81);
            box-shadow: 5px 5px 15px -6px rgba(0, 0, 0, 0.81);
        }
    }
`;

const PosterImgWrapper = styled.div`
    max-width: 50px;
    min-width: 50px;
    height: auto;
    border-radius: 2px;
    overflow: hidden;
`;

const PosterImg = styled.img`
    width: 100%;
`;
const MovieDetails = styled.div`
    display: flex;
    flex-direction: column;
    margin-left: 12px;
    text-align: left;
    & p {
        color: ${(p) => p.theme.colors.white};
    }
`;

const MovieTitle = styled.p`
    color: ${(p) => p.theme.colors.main};
    font-weight: 600;
    @media (min-width: 800px) {
        font-size: 18px;
    }
`;
const Year = styled.p`
    color: ${(p) => p.theme.colors.main};
    font-weight: 600;
    margin-top: 8px;
    @media (min-width: 800px) {
        font-size: 18px;
    }
`;

const Movie: FC<IMovieApi> = ({ Poster: poster, Title: title, Type: type, Year: year, imdbID: id }) => {
    return (
        <Container href={`https://www.imdb.com/title/${id}`} rel='noopener' target='__blank'>
            <PosterImgWrapper>
                <PosterImg src={poster} />
            </PosterImgWrapper>
            <MovieDetails>
                <MovieTitle>{title}</MovieTitle>
                <Year>{year}</Year>
            </MovieDetails>
        </Container>
    );
};

export default Movie;
