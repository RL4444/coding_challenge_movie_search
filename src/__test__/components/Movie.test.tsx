import React from 'react';
import { shallow } from 'enzyme';

import Movie from '../../components/Movie';
import { IMovieApi } from '../../types/movie';

const MOCK_DATA: IMovieApi = {
    Poster: 'A test String',
    Title: 'A test String',
    Type: 'A test String',
    Year: 'A test String',
    imdbID: 'A test String',
};

it('renders without crashing', () => {
    const { Poster, Title, Type, Year, imdbID } = MOCK_DATA;

    const component = shallow(<Movie Poster={Poster} Title={Title} Type={Type} Year={Year} imdbID={imdbID} />);

    expect(component).toMatchSnapshot();
});
