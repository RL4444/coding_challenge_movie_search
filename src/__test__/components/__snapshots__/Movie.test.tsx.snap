// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<styled.a
  href="https://www.imdb.com/title/A test String"
  rel="noopener"
  target="_blank"
>
  <styled.div>
    <styled.img
      src="A test String"
    />
  </styled.div>
  <styled.div>
    <styled.p>
      A test String
    </styled.p>
    <styled.p>
      A test String
    </styled.p>
  </styled.div>
</styled.a>
`;
